sql/2021/privacy/purposes_for_iab_tcf_v2.sql

```diff
@@ -0,0 +1,111 @@
+#standardSQL
+# Counts of purpose declarations on websites using IAB Transparency & Consent Framework
+# cf. https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#tcdata
+
+# https://stackoverflow.com/a/65054751/7391782
+# Warning: fails if there are colons in the keys/values, but these are not expected
+CREATE TEMPORARY FUNCTION ExtractKeyValuePairs(input STRING)
+RETURNS Array<STRUCT<key String,value String>>
+AS
+((select
+  array(
+    select as struct 
+      trim(split(kv, ':')[SAFE_OFFSET(0)]) as key, 
+      trim(split(kv, ':')[SAFE_OFFSET(1)]) as value
+    from t.kv kv
+  ) 
+from unnest([struct(split(translate(input, '{}"', '')) as kv)]) t
```

Comment:
Looks like the linter doesn't support this syntax. So tell it to ignore it (plus fixing the capitalisation issues at the same time!):

```suggestion
FROM UNNEST([STRUCT(SPLIT(TRANSLATE(input, '{}"', '')) AS kv)]) t -- noqa: PRS
```

Issue raised here: https://github.com/sqlfluff/sqlfluff/issues/1271 and fix has been merged so will be in the next release of the linter, but for now do the above.